File: backend/app/schemas/enquiries.py
"""Schemas for onboarding enquiries."""
from __future__ import annotations

from datetime import datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel, ConfigDict, EmailStr, Field


class EnquiryBase(BaseModel):
    customer_type: str = Field(..., pattern="^(individual|business)$")
    full_name: str
    email: EmailStr
    phone: str
    company_name: Optional[str] = None
    fleet_size: Optional[str] = None
    operating_area: Optional[str] = None
    preferred_contact_method: Optional[str] = None
    expected_go_live_date: Optional[datetime] = None
    tracking_use_case: Optional[str] = None
    hardware_choices: List[str]
    add_ons: List[str] = []
    message: Optional[str] = None
    terms_accepted: bool
    privacy_accepted: bool


class EnquiryCreate(EnquiryBase):
    pass


class EnquiryUpdate(BaseModel):
    status: Optional[str] = Field(default=None, pattern="^(new|quoted|awaiting_payment|onboarded|closed_lost)$")
    quoted_monthly: Optional[float] = None
    quoted_hardware_total: Optional[float] = None
    quote_sent_at: Optional[datetime] = None
    responded_at: Optional[datetime] = None
    closed_at: Optional[datetime] = None
    admin_notes: Optional[str] = None


class EnquiryResponse(EnquiryBase):
    id: UUID
    status: str
    quoted_monthly: Optional[float] = None
    quoted_hardware_total: Optional[float] = None
    quote_sent_at: Optional[datetime] = None
    responded_at: Optional[datetime] = None
    closed_at: Optional[datetime] = None
    admin_notes: Optional[str] = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: client-web/src/lib/api/enquiries.ts
import { frappeCall } from "$lib/api/frappe";

export type EnquiryPayload = {
  full_name: string;
  email: string;
  phone: string;
  company_name?: string | null;
  fleet_size?: string | null;
  operating_area?: string | null;
  preferred_contact_method?: string | null;
  expected_go_live_date?: string | null;
  tracking_use_case?: string | null;
  hardware_choices: string[];
  add_ons: string[];
  fleet_segments: Array<{ vehicle_type: string; label: string; count: number }>;
  message?: string | null;
  terms_accepted: boolean;
  privacy_accepted: boolean;
};

export async function submitEnquiry(payload: EnquiryPayload) {
  return frappeCall<{ received: boolean; reference: string; message: string }>(
    "omni_operations.customer_portal.enquiries.submit_quote_request",
    { method: "POST", body: payload },
  );
}
